tests/no-striping.cpp

```diff
@@ -0,0 +1,232 @@
+                CHECK(cur->shape.dims.width ==
+                      props.video[0].camera.settings.shape.x);
+                CHECK(cur->shape.dims.height ==
+                      props.video[0].camera.settings.shape.y);
+                ++nframes;
+            }
+            {
+                uint32_t n = consumed_bytes(beg, end);
+                OK(acquire_unmap_read(runtime, 0, n));
+                if (n)
+                    LOG("stream %d consumed bytes %d", 0, n);
+            }
+            clock_sleep_ms(&throttle, 100.0f);
+
+            LOG(
+              "stream %d nframes %d time %f", 0, nframes, clock_toc_ms(&clock));
+        } while (DeviceState_Running == acquire_get_state(runtime) &&
+                 nframes < props.video[0].max_frame_count);
+
+        OK(acquire_map_read(runtime, 0, &beg, &end));
+        for (cur = beg; cur < end; cur = next(cur)) {
+            LOG("stream %d counting frame w id %d", 0, cur->frame_id);
+            CHECK(cur->shape.dims.width ==
+                  props.video[0].camera.settings.shape.x);
+            CHECK(cur->shape.dims.height ==
+                  props.video[0].camera.settings.shape.y);
+            ++nframes;
+        }
+        {
+            uint32_t n = consumed_bytes(beg, end);
+            OK(acquire_unmap_read(runtime, 0, n));
+            if (n)
+                LOG("stream %d consumed bytes %d", 0, n);
+        }
+
+        CHECK(nframes == props.video[0].max_frame_count);
+    }
+
+    OK(acquire_stop(runtime));
+}
+
+int
+main()
+{
+    auto runtime = acquire_init(reporter);
+    acquire(runtime, TEST ".zarr");
+
+    CHECK(fs::is_directory(TEST ".zarr"));
+
+    const auto external_metadata_path =
+      fs::path(TEST ".zarr") / "0" / ".zattrs";
+    CHECK(fs::is_regular_file(external_metadata_path));
+    ASSERT_GT(size_t, "%zu", fs::file_size(external_metadata_path), 0);
+
+    const auto group_zattrs_path = fs::path(TEST ".zarr") / ".zattrs";
+    CHECK(fs::is_regular_file(group_zattrs_path));
+    ASSERT_GT(size_t, "%zu", fs::file_size(group_zattrs_path), 0);
+
+    const auto zarray_path = fs::path(TEST ".zarr") / "0" / ".zarray";
+    CHECK(fs::is_regular_file(zarray_path));
+    ASSERT_GT(size_t, "%zu", fs::file_size(zarray_path), 0);
+
+    // check metadata
+    std::ifstream f(zarray_path);
+    json zarray = json::parse(f);
+
+    auto shape = zarray["shape"];
+    ASSERT_EQ(int, "%d", expected_frames_per_chunk, shape[0]);
+    ASSERT_EQ(int, "%d", 1, shape[1]);
+    ASSERT_EQ(int, "%d", frame_height, shape[2]);
+    ASSERT_EQ(int, "%d", frame_width, shape[3]);
+
+    auto chunks = zarray["chunks"];
+    ASSERT_EQ(int, "%d", expected_frames_per_chunk, chunks[0]);
+    ASSERT_EQ(int, "%d", 1, chunks[1]);
+    ASSERT_EQ(int, "%d", frame_height, chunks[2]);
+    ASSERT_EQ(int, "%d", frame_width, chunks[3]);
+
+    // check chunked data
+    auto chunk_size = chunks[0].get<int>() * chunks[1].get<int>() *
```

Comment:
trying to make sure I understand the logic behind the test - if the shape and chunk metadata the striping is fixed?

Reply:
Honestly, the test itself doesn't demonstrate anything. It generates data for a visual inspection. Without knowing `t` in the [expression](https://github.com/acquire-project/acquire-driver-common/blob/main/src/simcams/imfill.pattern.cpp#L54)

```c
127.0f * (sinf(6.28f * (mid * 10.0f + (dx2 + dy2) * 1e-2f)) + 1.0f);
```

that generates each data point, that's the best I can do without doing, say, nonlinear least squares in the test itself to find it.

Reply:
Honestly, the test itself doesn't demonstrate anything. Given the [expression](https://github.com/acquire-project/acquire-driver-common/blob/main/src/simcams/imfill.pattern.cpp#L54) that generates the radial sine pattern

```cpp
(T)(127.0f * (sinf(6.28f * (t * 10.0f + (dx2 + dy2) * 1e-2f)) + 1.0f));
```

I had been hoping to find `t` numerically from the data but without a good starting point my methods are failing me. I move we just get rid of the test.